### PR TITLE
CanonicalSerializer support for ed25519 and x25519

### PR DESCRIPTION
**Pull request by kchalkias**
## Motivation

Adding support for canonical serialization in `ed25519` (public key and signature) and `x25519` (static public key). Also some function reordering to maintain function order (to match trait's functions order).

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

YES

## Test Plan

Existing

### CODE REVIEW

File: common/canonical_serialization/src/simple_deserializer.rs [hunk 1]
```diff
@@ -59,26 +59,6 @@ impl<'a> CanonicalDeserializer for SimpleDeserializer<'a> {
         Ok(b != 0)
     }
 
-    fn decode_btreemap<K: CanonicalDeserialize + std::cmp::Ord, V: CanonicalDeserialize>(
```
Comment (weilei0107) [inline common/canonical_serialization/src/simple_deserializer.rs]: Why is this file changed? It seems like code got reorganized. Could you revert it?

File: common/canonical_serialization/src/simple_serializer.rs [hunk 2]
```diff
@@ -58,41 +58,6 @@ where
         Ok(self)
     }
 
-    fn encode_tuple_iterator<K: CanonicalSerialize, V: CanonicalSerialize, I>(
```
Comment (weilei0107) [inline common/canonical_serialization/src/simple_serializer.rs]: Similarly this file shouldn't need to change.

File: common/canonical_serialization/src/simple_deserializer.rs (hunk 1 above)
Comment (kchalkias) [reply common/canonical_serialization/src/simple_deserializer.rs]: @weilei0107 it's on purpose, just took the opportunity to improve readability and consistency (implementation order to match function definition order on the trait).

Comment (davidiw) [reply common/canonical_serialization/src/simple_deserializer.rs]: Perhaps we should have fixed the definition within the trait. Now this is no longer alphabetically ordered :(

Comment (kchalkias) [reply common/canonical_serialization/src/simple_deserializer.rs]: @davidiw good point, but I guess there're two different opinions here: "alphabetical" vs "conceptual". 
For instance, the creator of a `trait` might decided that numbers should be grouped even if alphabetically there is some distance. Thus, 
`decode` i8, i32, i64, i128, u8, u32, u64, u128, string 
might be better conceptually than the alphabetical ordering 
`decode` i128, i8, i32, i64, string, u128, u8, u32, u64
